core/src/test/java/org/apache/iceberg/TestManifestWriterVersions.java

```diff
@@ -136,24 +141,31 @@ public void testV2ManifestRewriteWithInheritance() throws IOException {
     checkRewrittenManifest(manifest2, SEQUENCE_NUMBER, 0L);
 
     // should not inherit the v2 sequence number because it was written into the v2 manifest
-    checkRewrittenEntry(readManifest(manifest2), 0L);
+    checkEntry(readManifest(manifest2).asEntry(), ManifestEntry.Status.EXISTING, 0L);
+    checkDataFile(readManifest(manifest2), FileStatus.EXISTING, 0L);
   }
 
-  void checkEntry(ManifestEntry entry, Long expectedSequenceNumber) {
-    Assert.assertEquals("Status", ManifestEntry.Status.ADDED, entry.status());
+  void checkEntry(ManifestEntry entry, ManifestEntry.Status status, Long expectedSequenceNumber) {
```

Comment:
This changes how `checkEntry` and `checkDataFile` are used. This updates `checkEntry` to now validate the v1 view of the data by inspecting the `ManifestEntry` and the `DataFile` it wraps. Similarly, `checkDataFile` validates the v2 view where all fields are part of `DataFile`.